### ISSUE TITLE
Kubernetes V2 dockerImageReplacer may eat up all the CPU

### ISSUE THREAD

**Issue by eisig** [spinnaker/spinnaker#2742]
### Issue Summary:
A invalid docker image,  may cause the clouddriver to eat up all the CPU.

 I deploy a k8s  manifest with a invalid container image, but it is valid for k8s.
Every time I view the application In the UI, the clouddriver eat up the CPU for a long. And the `applications/{xxx}/serverGroups ` return 500.

The sample code below shows that `getNameFromReference` will not return for a long time.
```
    ArtifactReplacer.Replacer replacer = ArtifactReplacerFactory.dockerImageReplacer();
    String image = "xxx.dkr.ecr.cn-north-1.amazonaws.com.cn/repo1/abc-cif:%!s(float64=1)";
    String name =  replacer.getNameFromReference(image);
```

Do we really need that complex regex to validate the image here? Can we just use a simple regex like ```"^([^:]+)((?::.*)?)$"``` to split the image?
### Cloud Provider(s):
Kubernetes V2

**Comment by Sgydoo** [comment]
I found that having this line in my Manifest caused Clouddriver's _getNameFromReference_ method to go into an infinite loop:

```xxxxxxx.dkr.ecr.eu-west-1.amazonaws.com/${trigger.properties['IMAGE_NAME']}:${trigger.properties['VERSION']}```

I suspect the pattern-matching characters in that string are causing problems for _java.util.regex.Pattern_.

Should those properties have been interpolated before the pattern match is attempted?

**Comment by lwander** [comment]
Can you confirm that those properties are being sent to clouddriver uninterpreted? They should have been evaluated before requesting the deployment operation.

**Comment by Sgydoo** [comment]
Initially, the deployment is working fine, so they must have been evaluated correctly at some point. The deployment triggers this infinite looping behaviour.

Once Clouddriver is in the loop, I can't do anything more with it.

**Comment by Sgydoo** [comment]
Question... Are we supposed to be able to use trigger properties and parameters in our Manifests, or have I misunderstood the use case?

**Comment by Sgydoo** [comment]
To clarify, I can confirm that Clouddriver sees the uninterpreted properties. I started Clouddriver with the JDWP agent and connected to the agent using Eclipse debugging tools. The string I mentioned above was extracted from the depths of a _java.util.regex.Pattern_ loop.

**Comment by lwander** [comment]
Yup trigger properties most certainly can be used in those manifests. In any case, the above linked PR will prevent clouddriver from attempting to parse all of these images.

**Comment by lwander** [comment]
It's also patched into 1.8, will come with 1.8.1 this week. https://github.com/spinnaker/clouddriver/commit/8f6ffbf7a8ae5491f0978363e7efd2d83c71e267

**Comment by Sgydoo** [comment]
Yay! Thanks :)
